Panic on PullSubscribe with missing permission

## Defect

Make sure that these boxes are checked before submitting your issue -- thank you!

 - [x] Included nats.go version
 - [ ] Included a [Minimal, Complete, and Verifiable example] (https://stackoverflow.com/help/mcve)

#### Versions of `nats.go` and the `nats-server` if one was involved:

- nats.go: 1.25
- nats-server: 2.9.0

#### OS/Container environment:

Docker

#### Steps or code to reproduce the issue:

PullSubscribe with missing permission produces this error on NATS side `Publish Violation - User "xxxxx", Subject "$JS.API.STREAM.PURGE.xxxxxx"` but a crash on NATS Go client.

#### Expected result:

Proper missing permission error

#### Actual result:

```
panic: runtime error: invalid memory address or nil pointer dereference[signal SIGSEGV: segmentation violation code=0x1 addr=0x1a0 pc=0x9dfee7]goroutine 37219 [running]:github.com/nats-io/nats%2ego.(*js).subscribe(0xc00051c420 - , {0xc0750b4600, 0x5b}, {0x0, 0x0}, 0x0, 0xc0750a8480, 0x1, 0x1, {0xc0750c4040, ...})/go/pkg/mod/github.com/nats-io/nats.go@v1.25.0/js.go:1721 +0x12c7github.com/nats-io/nats%2ego.(*js).PullSubscribe(0xc00051c420, {0xc0750b4600, 0x5b}, {0x0, 0x0}, {0xc0750c4040, 0x2, 0x2})/go/pkg/mod/github.com/nats-io/nats.go@v1.25.0/js.go:1289 +0x17a
```

@piotrpio Bumping this, has there been any more updates on this?

I've seen this issue after pull subscribing with non unique filtered consumer. We would expect this to just return an error but not panic.

client: `1.24.0`
server: `2.9.15`

```
err=nats: filtered consumer not unique on workqueue stream
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x1a0 pc=0xdcc48f]

goroutine 14161 [running]:
github.com/nats-io/nats%2ego.(*js).subscribe(0xc0011c2960, {0xc0005ec0a0, 0x4c}, {0x0, 0x0}, 0x0, 0xc0011c2e40, 0x1, 0x1, {0xc00105dc00, ...})
	/home/users/eslack/go/pkg/mod/github.com/nats-io/nats.go@v1.24.0/js.go:1719 +0x126f
github.com/nats-io/nats%2ego.(*js).PullSubscribe(0xc0011c2960, {0xc0005ec0a0, 0x4c}, {0xc000c384e0, 0x58}, {0xc00119e1e0, 0x1, 0x1})
	/home/users/eslack/go/pkg/mod/github.com/nats-io/nats.go@v1.24.0/js.go:1289 +0x18a
```

--- a/js.go
+++ b/js.go
@@ -1947,7 +1947,6 @@ func (js *js) subscribe(subj, queue string, cb MsgHandler, ch chan *Msg, isSync,
 		} else if consName == "" {
 			consName = getHash(nuid.Next())
 		}
-		var info *ConsumerInfo
 		if o.ctx != nil {
 			info, err = js.upsertConsumer(stream, consName, ccreq.Config, Context(o.ctx))
 		} else {
@@ -1959,7 +1958,11 @@ func (js *js) subscribe(subj, queue string, cb MsgHandler, ch chan *Msg, isSync,
 				cleanUpSub()
 				return nil, err
 			}
-			if consumer == _EMPTY_ ||
+			lookupConsumer := consumer
+			if lookupConsumer == _EMPTY_ {
+				lookupConsumer = consName
+			}
+			if lookupConsumer == _EMPTY_ ||
 				(apiErr.ErrorCode != JSErrCodeConsumerAlreadyExists && apiErr.ErrorCode != JSErrCodeConsumerNameExists) {
 				cleanUpSub()
 				if errors.Is(apiErr, ErrStreamNotFound) {
@@ -1972,7 +1975,7 @@ func (js *js) subscribe(subj, queue string, cb MsgHandler, ch chan *Msg, isSync,
 				cleanUpSub()
 			}
 
-			info, err = js.ConsumerInfo(stream, consumer)
+			info, err = js.ConsumerInfo(stream, lookupConsumer)
 			if err != nil {
 				return nil, err
 			}
@@ -2007,8 +2010,17 @@ func (js *js) subscribe(subj, queue string, cb MsgHandler, ch chan *Msg, isSync,
 				hasHeartbeats = info.Config.Heartbeat > 0
 			}
 		} else {
+			if info == nil {
+				cleanUpSub()
+				return nil, ErrConsumerCreationResponseEmpty
+			}
 			// Since the library created the JS consumer, it will delete it on Unsubscribe()/Drain()
 			sub.mu.Lock()
+			if sub.jsi == nil {
+				sub.mu.Unlock()
+				cleanUpSub()
+				return nil, ErrConsumerCreationResponseEmpty
+			}
 			sub.jsi.dc = true
 			sub.jsi.pending = info.NumPending + info.Delivered.Consumer
 			// If this is an ephemeral, we did not have a consumer name, we get it from the info
@@ -2020,6 +2032,10 @@ func (js *js) subscribe(subj, queue string, cb MsgHandler, ch chan *Msg, isSync,
 				}
 			}
 			sub.mu.Unlock()
+		}
+		if info == nil {
+			cleanUpSub()
+			return nil, ErrConsumerCreationResponseEmpty
 		}
 		// Capture max ack pending from the info response here which covers both
 		// success and failure followed by consumer lookup.

--- a/js.go
+++ b/js.go
@@ -2010,17 +2010,8 @@ func (js *js) subscribe(subj, queue string, cb MsgHandler, ch chan *Msg, isSync,
 				hasHeartbeats = info.Config.Heartbeat > 0
 			}
 		} else {
-			if info == nil {
-				cleanUpSub()
-				return nil, ErrConsumerCreationResponseEmpty
-			}
 			// Since the library created the JS consumer, it will delete it on Unsubscribe()/Drain()
 			sub.mu.Lock()
-			if sub.jsi == nil {
-				sub.mu.Unlock()
-				cleanUpSub()
-				return nil, ErrConsumerCreationResponseEmpty
-			}
 			sub.jsi.dc = true
 			sub.jsi.pending = info.NumPending + info.Delivered.Consumer
 			// If this is an ephemeral, we did not have a consumer name, we get it from the info
@@ -2032,10 +2023,6 @@ func (js *js) subscribe(subj, queue string, cb MsgHandler, ch chan *Msg, isSync,
 				}
 			}
 			sub.mu.Unlock()
-		}
-		if info == nil {
-			cleanUpSub()
-			return nil, ErrConsumerCreationResponseEmpty
 		}
 		// Capture max ack pending from the info response here which covers both
 		// success and failure followed by consumer lookup.

